torch/csrc/jit/tensorexpr/ir.h

```diff
@@ -733,6 +701,61 @@ class CallNode : public ExprNode<Op, BaseCallNode> {
   using BaseClass::BaseClass;
 };
 
+class TORCH_API CompareSelect : public ExprNode<CompareSelect> {
+ public:
+  CompareSelectOperation compare_select_op() const {
+    return compare_op_;
+  }
+  const Expr& lhs() const {
+    return this->lhs_;
+  }
+  const Expr& rhs() const {
+    return this->rhs_;
+  }
+  const Expr& ret_val1() const {
+    return this->ret_val1_;
+  }
+  const Expr& ret_val2() const {
+    return this->ret_val2_;
+  }
+
+  static Expr make(
+      const Expr& lhs,
+      const Expr& rhs,
+      CompareSelectOperation cmp_op) {
+    return Expr(
+        new CompareSelect(lhs, rhs, IntImm::make(1), IntImm::make(0), cmp_op));
+  }
+
+  static Expr make(
+      const Expr& lhs,
+      const Expr& rhs,
+      const Expr& ret_val1,
+      const Expr& ret_val2,
+      CompareSelectOperation cmp_op) {
+    return Expr(new CompareSelect(lhs, rhs, ret_val1, ret_val2, cmp_op));
+  }
+
+ private:
+  Expr lhs_;
+  Expr rhs_;
+  Expr ret_val1_;
+  Expr ret_val2_;
+  CompareSelectOperation compare_op_;
+  CompareSelect(
```

Comment:
You probably want to check that the types and lanes match between `lhs_` and `rhs_`, and between `ret_val1_` and `ret_val2_`.